src/Bridge/Symfony/Bundle/Test/ApiTestAssertionsTrait.php

```diff
@@ -95,14 +98,20 @@ public static function assertArraySubset($subset, $array, bool $checkForObjectId
     }
 
     /**
-     * @param array|string $jsonSchema
+     * @param Schema|array|string $jsonSchema
      */
     public static function assertMatchesJsonSchema($jsonSchema, ?int $checkMode = null, string $message = ''): void
     {
         $constraint = new MatchesJsonSchema($jsonSchema, $checkMode);
+
         static::assertThat(self::getHttpResponse()->toArray(false), $constraint, $message);
     }
 
+    public static function assertMatchesResourceJsonSchema(string $resourceClass, string $format = 'json', bool $output = true, ?string $operationType = null, ?string $operationName = null, ?Schema $schema = null, ?array $serializerContext = null, bool $forceCollection = false): void
```

Comment:
This seems like a nightmarish assertion method. :see_no_evil:

Probably not a good idea. It'll be so unreadable.

Reply:
Could just be `string $resourceClass, string $format = 'json', bool $output = true, ?string $operationType = null, ?string $operationName = null` IMHO.

Also, we should rename `output` in `response`.

Reply:
Do you want to read in your test:

```php
$this->assertMatchesResourceJsonSchema(Book::class, 'json', true, 'collection', 'post');
```

I'm sure I don't. It's completely ridiculous as an assertion method.

Reply:
:+1: what is output used for? 

We should try to move some usual arguments in the assertion function name then, something like `assertCollectionMatchesResourceJsonSchema`? 
Does it make sense to have another format then `json` to compare against a json schema?
Not sure to get how it differs by operation name (`post`, `get`) for the comparison.

Reply:
> what is output used for?

It's actually to choose the type of the schema, i.e. input schema or output schema.

Reply:
@dunglas I think it's better if we can change `SchemaFactory::buildSchema` to take a `string $type` argument instead, with constants for input and output.

Reply:
Okay then we'll have 3 arguments:
- format
- operation
- type (Output/Input constants defaults to Output)

Reply:
@teohhanhui LGTM! Can you provide open a PR? Or even add a commit to this one?